src/shared/filesystem.ts

```diff
@@ -6,102 +6,50 @@
 'use strict'
 
 import * as fs from 'fs'
+import * as os from 'os'
+import * as path from 'path'
+import { promisify } from 'util'
 
-export async function accessAsync(path: string | Buffer): Promise<void> {
-    await new Promise<void>((resolve, reject) => fs.access(path, err => {
-        if (!err) {
-            resolve()
-        } else {
-            reject(err)
-        }
-    }))
+const access = promisify(fs.access)
+
+export type PathLike = fs.PathLike
+
+export const accessAsync = async (pathLike: fs.PathLike): Promise<void> => {
+    return access(pathLike)
 }
 
-export async function mkdirAsync(
-    path: fs.PathLike,
-    options?: number | string | fs.MakeDirectoryOptions | undefined | null
-): Promise<void> {
-    await new Promise<void>((resolve, reject) => fs.mkdir(path, options, err => {
-        if (!err) {
-            resolve()
-        } else {
-            reject(err)
-        }
-    }))
+export const mkdirAsync = promisify(fs.mkdir)
+
+const mkdtemp = promisify(fs.mkdtemp)
+
+export const getTempDirPath = (prefix: string = 'vsctk') => {
+    return path.join(
+        os.type() === 'Darwin' ? '/tmp' : os.tmpdir(),
```

Comment:
What's the output of `os.tmpdir()` on a Mac? Could this potentially affect other Unix-like OSes?

Reply:
That was part of the issue I was resolving. See: https://github.com/aws/aws-toolkit-vscode/issues/240

I don't know what to expect on Linux but I'd prefer to use `os.tmpdir()` unless we have reason to believe it is a problem.